src/compute-client/src/plan.proto

```diff
@@ -76,6 +76,8 @@ message ProtoPlan {
    message ProtoPlanLetRec {
         repeated mz_expr.id.ProtoLocalId ids = 1;
         repeated ProtoPlan values = 2;
+        repeated uint64 max_iters = 4;
+        repeated bool max_iters_present = 5;
```

Comment:
I assume there is no `repeated optional uint64` that would make this second field unnecessary?

Relatedly, but not really relevant to this PR, I have been wondering why `LetRec` is represented with separate lists for ids and values (and now max_iters). ISTM that instead having a single `bindings` list with `(id, value, max_iter)` entries would be more convenient since it doesn't require ensuring that the lists have the same length all the time. But I'm probably missing something.

Reply:
We discussed that this might be a better representation but decided to not change it as part of the ongoing epic. I think it must be a skunkworks project or something similar, unless we find ~3-4 days of in-band work to do this as part of addressing tech debt.

Reply:
Makes sense, thanks!

Regarding `max_iters_present`, Alex mentioned in another comment:
> One general comment: we should absolutely prohibit people setting the MAX_ITERATIONS for some binding to 0, otherwise we run the risk of all sorts of incorrect optimizations for those blocks (or unnecessary complicated transform code).

Doing that would allow us to drop `max_iters_present` here and just use 0 to mean "no limit set".

Reply:
I'd prefer it, since it statically removes the possibility that the two lists might become inconsistent (e.g. have different lengths), so there is one thing less to check at runtime. But your argument about special values is valid too, so I don't oppose leaving things as they are.

Reply:
Changed it.